isototest/src/action.rs

```diff
@@ -77,73 +83,99 @@ pub async fn write_to_console(client: &VncClient, text: &str) -> Result<(), VncE
+        'A' => Some(KeyCode::A),
+        'B' => Some(KeyCode::B),
+        'C' => Some(KeyCode::C),
+        'D' => Some(KeyCode::D),
+        'E' => Some(KeyCode::E),
+        'F' => Some(KeyCode::F),
+        'G' => Some(KeyCode::G),
+        'H' => Some(KeyCode::H),
+        'I' => Some(KeyCode::I),
+        'J' => Some(KeyCode::J),
+        'K' => Some(KeyCode::K),
+        'L' => Some(KeyCode::L),
+        'M' => Some(KeyCode::M),
+        'N' => Some(KeyCode::N),
+        'O' => Some(KeyCode::O),
+        'P' => Some(KeyCode::P),
+        'Q' => Some(KeyCode::Q),
+        'R' => Some(KeyCode::R),
+        'S' => Some(KeyCode::S),
+        'T' => Some(KeyCode::T),
+        'U' => Some(KeyCode::U),
+        'V' => Some(KeyCode::V),
+        'W' => Some(KeyCode::W),
+        'X' => Some(KeyCode::X),
+        'Y' => Some(KeyCode::Y),
+        'Z' => Some(KeyCode::Z),
+        'a' => Some(KeyCode::a),
+        'b' => Some(KeyCode::b),
+        'c' => Some(KeyCode::c),
+        'd' => Some(KeyCode::d),
+        'e' => Some(KeyCode::e),
+        'f' => Some(KeyCode::f),
+        'g' => Some(KeyCode::g),
+        'h' => Some(KeyCode::h),
+        'i' => Some(KeyCode::i),
+        'j' => Some(KeyCode::j),
+        'k' => Some(KeyCode::k),
+        'l' => Some(KeyCode::l),
+        'm' => Some(KeyCode::m),
+        'n' => Some(KeyCode::n),
+        'o' => Some(KeyCode::o),
+        'p' => Some(KeyCode::p),
+        'q' => Some(KeyCode::q),
+        'r' => Some(KeyCode::r),
+        's' => Some(KeyCode::s),
+        't' => Some(KeyCode::t),
+        'u' => Some(KeyCode::u),
+        'v' => Some(KeyCode::v),
+        'w' => Some(KeyCode::w),
+        'x' => Some(KeyCode::x),
+        'y' => Some(KeyCode::y),
+        'z' => Some(KeyCode::z),
+        ' ' => Some(KeyCode::SPACE),
+        '!' => Some(KeyCode::ExcMrk),
+        '@' => Some(KeyCode::At),
+        '#' => Some(KeyCode::Pound),
+        '$' => Some(KeyCode::Dollar),
+        '%' => Some(KeyCode::Percent),
+        '^' => Some(KeyCode::Caret),
+        '&' => Some(KeyCode::And),
+        '*' => Some(KeyCode::Ast),
+        '(' => Some(KeyCode::LRBrace),
+        ')' => Some(KeyCode::RRBrace),
+        '-' => Some(KeyCode::Minus),
+        '_' => Some(KeyCode::UScore),
+        '=' => Some(KeyCode::Equals),
+        '+' => Some(KeyCode::Plus),
+        '[' => Some(KeyCode::LBracket),
+        ']' => Some(KeyCode::RBracket),
+        '{' => Some(KeyCode::LCrlBrace),
+        '}' => Some(KeyCode::RCrlBrace),
+        '\\' => Some(KeyCode::BckSlash),
+        '|' => Some(KeyCode::Pipe),
+        ';' => Some(KeyCode::SColon),
+        ':' => Some(KeyCode::Colon),
+        '\'' => Some(KeyCode::SQuote),
+        '"' => Some(KeyCode::DblQuote),
+        ',' => Some(KeyCode::Comma),
+        '.' => Some(KeyCode::Period),
+        '/' => Some(KeyCode::FwdSlash),
+        '<' => Some(KeyCode::LThan),
+        '>' => Some(KeyCode::GThan),
+        '?' => Some(KeyCode::Question),
         _ => None,
```

Comment:
Can we 
```suggestion
        _ => { panic!("KeyCode implementation missing {c}")},
```

Reply:
I would prefer not to panic unless absolutely necessary, but rather escalate an `Error` back to `testmaster` or whatever calls that so it can be handled there

Reply:
But instead of returning an `Option<u32>` we could return a `Result<u32, VncError>` which can then be escalated further up the chain by `write_to_console()`